### PR TITLE
Snomed to icd10

### PR DESCRIPTION
**Pull request by saeedbidi**
The following changes are made:

1. Null value list is completed for RawDataset
2. A method is added to processed dataset for mapping the SNOMED codes to ICD10 along with its test function
3. Tab format file support is added to test_raises_error_if_not_csv_or_feather_txt() (optional).
4. The date is now sorted in "test_load_processed_dataset" since in some runs the test function failed due to the random order of the table.

Note: we can perhaps change the path for the added mapping file or even the function.

### CODE REVIEW

File: tretools/datasets/processed_dataset.py [hunk 1]
```diff
@@ -84,5 +91,71 @@ def deduplicate(self, date_start: Optional[str] = None) -> ProcessedDataset:
...
         return processed_dataset
+    
+    def snomed_to_icd10(self, SNOMED_ICD10_map, dataset) -> ProcessedDataset:
+        '''
+        It maps SNOMED data to ICD10s using a mapping file.
+        Args:
+            SNOMED_ICD10_map: The mapping file
+            dataset (ProcessedDataset): The dataset for mapping
+        '''
+        processed_dataset = ProcessedDataset(path=self.path, dataset_type=self.dataset_type, coding_system=self.coding_system)
```
Comment (CarolineMorton) [inline tretools/datasets/processed_dataset.py:102]: I am confused as to what this is supposed to be doing here. You are crreating a ProcessedDataset that is the same the object that is calling this method. 

Also the SNOMED_ICD10_map is not used. I thik this method can be deleted.

File: examples/Demo.ipynb [hunk 2]
```diff
@@ -13,20 +17,16 @@
     "- Summary Report\n",
     "\n",
     "Codelists and Datasets form the building blocks of the system. They come together in specific queries in Phenotype Reports - to answer questions such as which patients in a given dataset have an event that matches a code in given codelist. A phenotype in this context is defined as a person who has a code in an identified codelist. A person might have several codes in 1 dataset, 1 qualifying code in different codelists or multiple codes in multiple codelists. "
```
Comment (CarolineMorton) [inline examples/Demo.ipynb:19]: Please get into the habit of not commiting this notebook.

File: tretools/datasets/base.py [hunk 3]
```diff
@@ -46,7 +46,8 @@ def _load_data(self, path: str) -> pl.DataFrame:
         if not self._check_path(path):
             raise DatasetPathNotCorrect(f"Invalid path for Dataset: {path}")
         
-        null_values = ["", " ", "NULL"]
+        null_values = ["", " ", "NULL", "NA", ".", "               ","                    ","6.92851E+14"]
```
Comment (CarolineMorton) [inline tretools/datasets/base.py:49]: This number can be removed.

File: tretools/datasets/processed_dataset.py [hunk 4]
```diff
@@ -84,5 +91,71 @@ def deduplicate(self, date_start: Optional[str] = None) -> ProcessedDataset:
...
+        temp_file = tempfile.NamedTemporaryFile(suffix=".csv",mode="w", delete=False)
+        writer = csv.DictWriter(temp_file, fieldnames=["code", "term"])
+        writer.writeheader()
+        for code, term in mapped_data.items():
+            writer.writerow({"code": code, "term": term})
+        temp_file.close()
+        # # make a new codelist object with the new data
+        new_codelist = ProcessedDataset(temp_file.name, "ICD10",coding_system ="SNOMED")
+        # new_codelist = Codelist(temp_file.name, "ICD10")
+        return new_codelist
```
Comment (CarolineMorton) [inline tretools/datasets/processed_dataset.py:158]: What is this?

**Review by CarolineMorton — CHANGES_REQUESTED** [review]

File: tretools/datasets/processed_dataset.py (hunk 1 above)
Comment (saeedbidi) [reply tretools/datasets/processed_dataset.py:102]: That was in my draft before I wrote `map_snomed_to_icd10`. I forgot to delete it, deleted now.

File: tretools/datasets/processed_dataset.py (hunk 4 above)
Comment (saeedbidi) [reply tretools/datasets/processed_dataset.py:158]: I wrote this method following the Asana task: "Add method to processed dataset to map the SNOMED codes to ICD10". It maps the SNOMED to ICD10. I can remove the comment `# new_codelist = Codelist(temp_file.name, "ICD10"`

**Comment by CarolineMorton** [comment]
Fixed in different pull request.

**Pull request closed without merging**